oauth2_http/javatests/com/google/auth/oauth2/DefaultCredentialsProviderTest.java

```diff
@@ -176,6 +166,63 @@ public void getDefaultCredentials_noCredentials_singleGceTestRequest() {
         ComputeEngineCredentials.MAX_COMPUTE_PING_TRIES);
   }
 
+  @Test
+  public void getDefaultCredentials_noCredentials_linuxNotGce() throws IOException {
+    TestDefaultCredentialsProvider testProvider = new TestDefaultCredentialsProvider();
+    testProvider.setProperty("os.name", "Linux");
+    String productFilePath = "/sys/class/dmi/id/product_name";
+    InputStream productStream = new ByteArrayInputStream("test".getBytes());
+    testProvider.addFile(productFilePath, productStream);
+
+    assertFalse(ComputeEngineCredentials.checkStaticGceDetection(testProvider));
+  }
+
+  @Test
+  public void getDefaultCredentials_static_linux() throws IOException {
+    TestDefaultCredentialsProvider testProvider = new TestDefaultCredentialsProvider();
+    testProvider.setProperty("os.name", "Linux");
+    String productFilePath = "/sys/class/dmi/id/product_name";
+    InputStream productStream = new ByteArrayInputStream("Googlekdjsfhg".getBytes());
+    testProvider.addFile(productFilePath, productStream);
+
+    assertTrue(ComputeEngineCredentials.checkStaticGceDetection(testProvider));
+  }
+
+  @Test
+  public void getDefaultCredentials_static_windows_skipsLinuxPath() throws IOException {
```

Comment:
How is it tested that linux path is skipped?

Reply:
because same setup with Linux OS version returns true.